source/postprocess/visualization/material_properties.cc

```diff
@@ -106,6 +112,36 @@ namespace aspect
 
         this->get_material_model().evaluate(in, out);
 
+        // We want to output material properties as they are used in the
+        // program during assembly. To do so, some of the material averaging
+        // modes require a quadrature object -- but we do not have this,
+        // all we have is the mapped quadrature points. As a consequence,
+        // only do the averaging for those modes that do not require a
+        // functional quadrature object. This means that we only
+        // do the wrong thing for the Q1 averaging where the difference
+        // between input and output of the averaging operation is generally
+        // small and probably not visible anyway.
+        //
+        // The average() function checks whether any quadrature object it uses
+        // has the correct size. So passing an invalid object carries little
+        // risk if the list of averaging modes that require a quadrature
+        // object expands: Every time we generate graphical output for
+        // a model that uses this kind of averaging, we will trigger an
+        // exception.
+        Quadrature<dim> quadrature;
```

Comment:
is this variable used? It seems you construct an unnamed quadrature in the call below.

Reply:
Ah, that's a left-over from a previous iteration. Let me remove it.